lib/ui/page/home/page/chat/widget/AudioPlayer.dart

```diff
@@ -0,0 +1,239 @@
+                ),
+              ),
+            ),
+          ),
+        ),
+        const SizedBox(width: 16),
+        SizedBox(
+          width: 300,
+          child: Column(
+            crossAxisAlignment: CrossAxisAlignment.start,
+            children: [
+              Text(
+                widget.friendly_name ?? "<no name>",
+                style: style.fonts.medium.regular.onBackground,
+                maxLines: 1,
+                overflow: TextOverflow.ellipsis,
+              ),
+              const SizedBox(height: 4),
+              SelectionContainer.disabled(
+                child: StyledCupertinoButton(
+                  padding: EdgeInsets.zero,
+                  label: 'Download',
+                  style: style.fonts.smaller.regular.primary,
+                  onPressed: () async {
+                    var bn = widget.friendly_name ?? "audio.mp3";
+                    final file = await PlatformUtils.saveTo(
+                      '${Config.downloads}/${bn}',
+                    );
+                    if (file != null) {
+                      MessagePopup.success(
+                          '${widget.path_or_url} downloaded to ${Config.downloads}/${bn}');
+                    }
+                  },
+                ),
+              ),
+            ],
+          ),
+        ),
+      ]),
+      Row(children: [
+        Expanded(
+          child: Obx(() => Slider(
+              value: _currentSliderValueMs.value,
+              max: _maxSliderValueMs.value,
+              label: _currentSliderValueMs.round().toString(),
+              onChanged: (double value) {
+                setState(() {
+                  _audioStream?.seek(value);
+                });
+              })),
+        ),
+        Obx(() => Text(
+          "${(_currentSliderValueMs/60000.0).floor().toString().padLeft(2, '0')}:${((_currentSliderValueMs % 60000.0) / 1000.0).floor().toString().padLeft(2, '0')}/"
+              "${(_maxSliderValueMs/60000.0).floor().toString().padLeft(2, '0')}:${((_maxSliderValueMs % 60000.0) / 1000.0).floor().toString().padLeft(2, '0')}",
+          style: style.fonts.medium.regular.onBackground,
+          maxLines: 1,
+          overflow: TextOverflow.ellipsis,
+        )),
+      ]),
+    ]);
+  }
+
+  void _playAudio() {
+    var asrc = widget.path != null ? AudioSource.file(widget.path!) : AudioSource.url(widget.url!);
+    _audioStream = AudioUtils.createPlayStream(asrc, loop: false, stop_others: true);
+    _audio = _audioStream?.beginPlay(
+        onData: (_) => {
+            _audioStream?.getDurationStream().listen((event) {
+            _maxSliderValueMs.value = event.inMilliseconds.toDouble();
+            }),
+            _audioStream?.getPositionStream().listen((event) {
+              _currentSliderValueMs.value = event.inMilliseconds.toDouble();
+            })
+        },
+        onDone: () => {
+          _stopAudio(external_call: true)
+        });
```

Comment:
https://dart.dev/effective-dart/style#consider-changing-your-code-to-make-it-more-formatter-friendly